batavia/core/version.js

```diff
@@ -36,11 +36,24 @@ version.version_id = function(str) {
 }
 
 version.earlier = function(str) {
+    if (constants.BATAVIA_MAGIC === undefined) {
+        throw new exceptions.BataviaError.$pyclass('Version test called before version defined.')
+    }
     return magic_map[constants.BATAVIA_MAGIC] < version.version_id(str)
 }
 
 version.later = function(str) {
+    if (constants.BATAVIA_MAGIC === undefined) {
+        throw new exceptions.BataviaError.$pyclass('Version test called before version defined.')
+    }
```

Comment:
Are earlier and later still being used anywhere? Is there any reason to keep them? Is there a need to have both "later" *and* "at_least"? (I agree at_least is a much easier to read name...)

Reply:
I was thinking about simultaneous in-flight PRs, but it looks pretty clean right now. I'll chuck them out now and simplify things.

Reply:
Hah. We introduced a bunch of new ones, obviously, in the 3.7 branch. Merging against that isn't sufficient to update those as well.